mock/world/mockESDTAccount.go

```diff
@@ -238,3 +242,70 @@ func (a *Account) GetTokenNames() ([][]byte, error) {
 
 	return tokenNames, nil
 }
+
+// MockESDTData groups together all instances of a token (same token name, different nonces).
+type MockESDTData struct {
+	TokenIdentifier []byte
+	Instances       []*esdt.ESDigitalToken
+	LastNonce       uint64
+	Roles           [][]byte
+}
+
+func getOrCreateMockESDTData(tokenName string, resultMap map[string]*MockESDTData) *MockESDTData {
+	resultObj := resultMap[tokenName]
+	if resultObj == nil {
+		resultObj = &MockESDTData{
+			TokenIdentifier: []byte(tokenName),
+			Instances:       nil,
+			LastNonce:       0,
+			Roles:           nil,
+		}
+		resultMap[tokenName] = resultObj
+	}
+	return resultObj
+}
+
+// GetFullMockESDTData returns the information about all the ESDT tokens held by the account.
+func (a *Account) GetFullMockESDTData() (map[string]*MockESDTData, error) {
+	resultMap := make(map[string]*MockESDTData)
+	for key := range a.Storage {
+		if IsTokenKey([]byte(key)) {
```

Comment:
Optionally declare `key` as `[]bytes` once, before the `if-else` block.
```suggestion
	for strKey := range a.Storage {
	        key := []byte(strKey)
		if IsTokenKey(key) {
```